### DISCUSSION TITLE
Why RISC-V compared to, for example, WASM?

### DISCUSSION THREAD

**Discussion by zicklag** [risc0/risc0#61] [Q&A] [answered]
Consider this a low-priority question for my own learning purposes, if you don't mind answering.

What motivated the decision to build risc0 on top of the RISC-V architecture, instead of WASM, or another architecture? Is RISC-V just a very simple architecture or something like that? I know it seems to be popular in microcontrollers.

**Comment by brianretford** [comment, marked as answer]
Basically, yes. Its instruction set is very simple and readily encodable in arithmetic constraints. We may choose to support any number of other architectures in the future.
